packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -629,6 +630,51 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection {
 		}
 	}
 
+	public get disposed() {
+		if (!(this._disposed || this.socket.connected)) {
+			this.logger.sendTelemetryEvent({
+				eventName: "ConnectionNotYetDisposed",
+				driverVersion: pkgVersion,
+				details: JSON.stringify({
+					...this.getConnectionDetailsProps(),
+				}),
+			});
+		}
+		return this._disposed;
+	}
+
+	private get connected(): boolean {
+		return !this.disposed && this.socket.connected;
```

Comment:
I think it's fine, but I'd add big comment here about expectations: this function will return false during the time we are establishing connection. The expectation is that it will be called only after connection is fully established. I.e. there should no way to submit an op while we are connecting, as connection object is not exposed to Loader layer until connection is established.

Reply:
Since we reuse the socket it could be connected during establishing connection too but I added comment. It is a private function and exposed api like submit/submitSignal will not be exposed until we return connection to loader and so it will not be called before that.